### ISSUE TITLE
Netty dependency conflict with Flink

### ISSUE THREAD

**Issue by allenlee820202** [dsngroup/broke#41]
Flink uses **Netty 4.0.27** while we used **Netty 4.1.16**. I found that some ```bytebuf``` functions cannot work on Flink standalone cluster when decoding the messages. 
However, local test still works.

The error message is like:
```
taskmanager_1  | 2017-12-15 08:28:29,250 ERROR org.dsngroup.broke.client.handler.MqttMessageHandler          - An exceptionCaught() event is fired: java.lang.NoSuchMethodError: io.netty.buffer.ByteBuf.readRetainedSlice(I)Lio/netty/buffer/ByteBuf;
```

**Comment by tz70s** [comment]
Can you downgrade version to `4.0.27`?

**Comment by allenlee820202** [comment]
A local test looks fine. Only a cluster running on containers doesn't work.
I'll try to downgrade first.

**Comment by tz70s** [comment]
I see. No, It’s because the oudated version in container, you should build your own container.

**Comment by tz70s** [comment]
Which image do you use? You can find out dockerfile  on docker hub, and modify to the matched version.

**Comment by allenlee820202** [comment]
I think it is not the problem of Docker. I used ```docker-compose``` provided by [Flink](https://github.com/apache/flink/tree/master/flink-contrib/docker-flink).

**Comment by allenlee820202** [comment]
In ```Netty 4.0```, ByteBuf doesn't have ```readRetainedSlice``` while ```Netty 4.1``` has. I think non-cluster tests don't show this problem because they didn't need ```Netty```.

**Comment by tz70s** [comment]
Ok, flink has an issue for not upgrading netty version, you can checkout at pom file on the repo. Anyway, go downgrade first.

**Comment by tz70s** [comment]
Fix in #42
